### ISSUE TITLE
Only works for public repositories due to lack of auth accessing GitHub API

### ISSUE THREAD

**Issue by nealeu** [green-coding-berlin/green-ci-branch-activity-checker#2]
`dateofcommit=$(curl -s https://api.github.com/repos/green-coding-berlin/eco-ci-activity-checker/commits/main)` only works because that repository is public.

For a private repository an auth token is needed

**Comment by ArneTR** [comment]
@nealeu Thanks for bringing this up!

@dan-mm Can you backport the fix from our other eco-ci activites to this one?

**Comment by dan-mm** [comment]
This ended up being a bit more complicated than just backporting the fix from the eco-ci-energy estimator, as the commits endpoint has different authentication than the workflow endpoints.

I was able to support this however by adding the ability to pass in a personal-access-token to the action for use with private repositories. The readme has been updated with information about this.

While touching this, I also made a few improvements based on what I learnt while working on the eco-ci-energy-estimation action, and now the workflow-id, repo, and branch are optional parameters. The user no longer needs to know their workflow-id, unless they want to check on a different workflow than the action is being called on
